zerver/lib/cache.py

```diff
@@ -441,9 +451,20 @@ def to_dict_cache_key_id(message_id, apply_markdown):
     # type: (int, bool) -> Text
     return u'message_dict:%d:%d' % (message_id, apply_markdown)
 
+# returns the unique hash key for messages which trigger slash commands. These messages are not recorded in
+# the database, hence message id is not available and `to_dict_cache_key` method fails. Thus this method
+# has been added to return a unique hash key using a random string.
+def to_dict_cache_key_id_for_messages_triggering_slash_commands():
```

Comment:
Thanks for clarifying, though I'm still a bit confused.

If slash command messages end up in some cache, would the cache key need to be deterministic (instead of randomly generated every time you call this function)?

Alternatively, if such messages never end up in a cache, should this just return `None`?